rocketpy/simulation/flight.py

```diff
@@ -996,8 +998,15 @@ def __simulate(self, verbose):
                                             [self.t, parachute]
                                         )
 
+                    # If controlled flight, post process must be done on sim time
+                    if self._controllers:
+                        phase.derivative(self.t, self.y_sol, post_processing=True)
+
         self.t_final = self.t
         self.__transform_pressure_signals_lists_to_functions()
+        if self._controllers:
+            # cache post process variables
+            self.__evaluate_post_process = np.array(self.__post_processed_variables)
```

Comment:
I know this question may sound you strange, but are you sure this is working?

I think the `Flight.__evaluate_post_process` is a property, so you should not be able to set its value. How is this working? 

Probably if you remove the leading underscores the whole code would break.

```suggestion
            self.__evaluate_post_process = np.array(self.__post_processed_variables)
```

Reply:
This makes it so the code inside `def __evaluate_post_process()` is skipped whenever that property is accessed anywhere else in the code. This is probably because of the `@cached_property` decorator. I don't know much else about it, but it works

Reply:
Ok, it seems like the cached decorator does not forbids us from settings.